packagegroup-ni-ptest-smoke: Remove unused ptests

usercw88 opened the pull request:
### Summary of Changes

Removed ptests that were found to be unused - ptests that were executed, but did not run any tests


### Justification

[AB#2755352](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2755352)
The testptest.py will be updated to assert if a ptest does not return any test results


### Testing
* [x] I have built the core package feed with this PR in place. (`bitbake packagefeed-ni-core`)
* [x] I have built the core package feed with this PR in place. (`bitbake packagegroup-ni-ptest-smoke`)
* [x] Executed ptests via debug run - all tests succeeded

### Procedure

* [x] I certify that the contents of this pull request complies with the [Developer Certificate of Origin](https://github.com/ni/nilrt/blob/HEAD/docs/CONTRIBUTING.md#developer-certificate-of-origin-dco).

Commit:
packagegroups: add pkggrp-ni-snac

Add a packagegroup to track package dependencies of the NILRT Secured,
Network-Attached Controller (SNAC) configuration.

Since the SNAC configuration is officially supported, add it to the core
package feed.

This packagegroup SHOULD NOT be installed to the NILRT runmode or
safemode images.

Signed-off-by: Alex Stewart <astewart.49c6@gmail.com>
(cherry picked from commit c683180cd7314c6150365e8d66f7ceee2c55da6c)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
pkggrp-ni-snac: add initial top level deps

Add some initial packages that we know are going to be used in the SNAC
v1.0 design.

Signed-off-by: Alex Stewart <astewart.49c6@gmail.com>
(cherry picked from commit 2b025f3ac13c78c9de97a99863b21943026efdcc)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
pkggrp-ni-snac: remove unneeded append assignment

The snac packagegroup is only expected to contain one logical subsection
of packages, so there is no need to split up the RDEPENDS assignments.
Use a single section.

Signed-off-by: Alex Stewart <astewart.49c6@gmail.com>
(cherry picked from commit fe72374f54d429305a388c25d98f3a1c5af0586a)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

texasaggie97 committed:
tmux-ipk: Add tmux configuration file to tmux IPK

* Create a directory for tmux configuration files in /usr/share/tmux/conf.d
* Add a /etc/tmux.conf file with default lock command
* Add a line to source all files in /usr/share/tmux/conf.d in /etc/tmux.conf
  * -q is to ignore errors if the directory is empty
* This allows snac mode to add a conf file to this location without modifying the main tmux.conf file

Signed-off-by: Mark Silva <mark.silva@ni.com>

tmux-bbappend: Add conf.d directory
(cherry picked from commit b09f844d9fc00ffd9be6bf5efed4714e646c6073)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

texasaggie97 committed:
libpwquality: Build recipe

Signed-off-by: Mark Silva <mark.silva@ni.com>
(cherry picked from commit 2fb42226c839e4835ce1788ea498cf4f5be450bb)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

texasaggie97 committed:
libpwquality: Add config file to recipe

* Add pwquality.conf that has the values that are required
  for SNAC mode
  * These values are more restrictive than the default
    values
* Add bbappend file to install the config file to the
  correct location

Signed-off-by: Mark Silva <mark.silva@ni.com>
(cherry picked from commit 94ffc6e845609209f14b28f279c7344867aa094b)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

AlexHearnNI committed:
libpam: enable pam-plugin-faillock when it's installed and customize settings

- update the pam-plugin-faillock package so that the plugin gets enabled when it's installed
- modify some faillock configuration settings
- prevent pam-plugin-faillock from being installed when ni-auth is installed

This change simplifies Secured, Network-Attached Controller (SNAC) configuration. faillock is required to be enabled on a SNAC. The faillock settings were chosen to comply with SNAC requirements. The conflict with ni-auth was added because from testing it appears that the faillock plugin is incompatible with the ni-auth plugin.

Signed-off-by: Alex Hearn <alex.hearn@ni.com>
(cherry picked from commit eeb7b911a79c4ef474e9de590fed11cde42749a0)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

rtollert committed:
packagegroup-ni-runmode: add nftables

nftables is the modern packet filtering solution on Linux. The nftables
userspace can be installed alongside iptables, but in general, only one can be
enabled at once. This commit adds the userspace but does not do anything with
the present iptables-based firewall configuration.

At present, there are no plans to introduce nftables into safemode, so add it
to the runmode packagegroup, not base.

Signed-off-by: Rich Tollerton <rich.tollerton@ni.com>
(cherry picked from commit 3718278554e32d6ae125012c4c2d0e212765954a)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

rtollert committed:
packagegroup-ni-snac: add firewalld

firewalld is the best-maintained high-level firewall administration tool on
Linux. We ultimately wish to replace our present direct use of iptables (via
initscript) with firewalld; but at present the existing configuration is
unchanged.

Signed-off-by: Rich Tollerton <rich.tollerton@ni.com>
(cherry picked from commit b03a9498821262a2015b4cbae0dc0ded0ec5553b)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

AlexHearnNI committed:
sudo: give the sudo group permission to run the sudo command

Signed-off-by: Alex Hearn <alex.hearn@ni.com>
(cherry picked from commit 6633023ce98e03edfb9b474d44d4278eef08b0b4)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
recipes-ni: add nilrt-snac recipe

nilrt-snac is an NI configuration tool for NILRT that allows a system
administrator to semi-automate the process of configuring the system
into the Secured, Network-Attached Controller (SNAC) configuration.

Add a recipe for it.

This recipe is NILRT-specific.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 50e89ba291c369f85be2a1cb9620c4d9653f610c)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
pkggrp-ni-snac: add nilrt-snac

Add the nilrt-snac configuration tool to the SNAC packagegroup to ensure
that it is always built into the core packagefeed. It should not be
installed to the base system image by default.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit dcc67efb8042dce97e786365db59327f2ee58814)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
nilrt-snac: add integration ptests

The nilrt-snac project now has an integration test suite. Add a
run-ptest entrypoint and pytest subpackage to run it.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 889156fca57e02c42cb2dd7c64c169527c44e3a9)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
pkggrp-ni-smoke: use tabs for indentation

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 4713fbdaf6e8efd79402496ce061075fba4c6a60)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
pkggrp-ni-ptest-smoke: add nilrt-snac-ptest

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 80eff37f7df9e4d7a158bca5841ae0222cc981a6)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
nilrt_snac: remove unneeded RDEPENDS

The ptest.bbclass already adds an RDEPENDS on `nilrt-snac`, but it is
overriden by the nilrt-snac-ptest RDEPENDS bb assignment.

Instead, append the ptest RDEPENDS.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 052582bfa03709f1b9d365e3ca03c9049291a170)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
nilrt-snac: make test output ptest-compliant

The NI ptest-parser (and ptests in general) use the AutoMake style of
test output. The nilrt-snac integration tests uses pytest formatting,
which is generally non-compliant and obscures individual testcase output
from being parsed by the RTOS ptesting pipeline.

Use the python3-unittest-automake-output plugin for pytest, to output
the test results in format ptest can parse.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 1c994aecb910d4a50a3928819f17c13961247f95)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
nilrt-snac: insert nirtcfg bin loc to PATH

The nirtcfg utility is installed to a non-standard path, for no good
reason. When running the nilrt-snac ptests from an interactive shell,
this nonstandard path is already searched. But when the ptests are run
from an non-interactive shell, they fail to resolve the path.

Add the nonstandard location to the search PATH.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 4ffacf69fb24b471804c6c692b6ade746b852919)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

amstewart committed:
nilrt-snac: fixup whitespace in nilrt-snac ptest

Python files use spaces for indentation.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 626231b2ead848b49c07a76d48631bdc1aeff5ba)
Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

usercw88 committed:
packagegroup-ni-ptest-smoke: Removing ptests from packagegroup

Remove ptests from packagegroup that do not report results
using the standard PASS/FAIL convention or require special
permission for execution

Signed-off-by: Can Wong <can.wong@ni.com>

jpautler commented:
Can you change the commit message to reflect that we are removing these ptests from the smoke test packagegroup because they do not report results using the standard ptest PASS/FAIL conventions and therefore our test framework is currently unable to parse individual test results from them? Also, can you remove the TODO in the Testing section and describe your testing?

chaitu236 committed:
linux-nilrt: Set CVE_VERSION to correct value

When CVE_VERSION is not explicitly set, it defaults to PV which does not
contain minor version number. This results in incorrect CVE status
determination.

So set CVE_VERSION to full version.

Signed-off-by: Chaitanya Vadrevu <chaitanya.vadrevu@ni.com>

rtollert committed:
issue*: delete NIAuth references

Telling users to "Log in with your NI-Auth credentials" doesn't really mean
anything to them. NIAuth is only barely documented to begin with (which says a
lot about how user-visible it was mean to be in the first place). Even if/when
niauth is disabled and passwd takes over, there's no real value to be gained by
disclosing this. Delete.

Signed-off-by: Richard Tollerton <rich.tollerton@ni.com>

jpautler commented:
Sorry, I missed this the first time...can you remove the commented block at the bottom where you moved the "removed" ptests. I don't think we need to keep a commented list. We can just review the commit history if we want to see what was removed.

One other nit- I wouldn't have the body of the commit be a continuation of the subject line. Make the subject line stand alone (packagegroup-ni-ptest-smoke: Remove ptests that are not generating results) and then have the body explain the details of why no results are being generated.

Otherwise, LGTM.

chaitu236 reviewed (changes requested):
> [Merge branch 'nilrt/master/kirkstone' into dev/kirkstone/remove_unused_ptest]

Our workflow for getting latest changes into in-development branch is to rebase instead of merge.
